### DISCUSSION TITLE
Rendering a form inside the context menu

### DISCUSSION THREAD

**Discussion by erikash** [icflorescu/mantine-contextmenu#181] [Q&A]
**Is your feature request related to a problem? Please describe.**
I would like to render a form inside the context menu as in:

Ideally it would support "sub-menus" similar to the attached screen recording:

![context menu](https://github.com/user-attachments/assets/ce490a80-cc99-4aee-affd-eb8e6a4eca42)

**Describe the solution you'd like**
In the docs i only see the following props - none of which seems applicable
```
**
       * Optional context menu item icon.
       */
      icon?: React.ReactNode;

      /**
       * Optional context menu item icon for the right side of the title.
       * If provided this will overwrite the submenu's ▶ icon
       */
      iconRight?: React.ReactNode;

      /**
       * Optional context menu item title.
       * If not provided, one will be generated automatically by "humanizing" the key.
       * @default humanize(key)
       */
      title?: React.ReactNode;
```

Any workarounds? 
other ideas?

Amazing job on this library and mantine-datatable!

Erik.

**Comment by icflorescu** [comment]
Yours is a very particular use-case that's kind of outside the scope of this project, but have a look at https://icflorescu.github.io/mantine-contextmenu/examples/custom-content/ for a starter to achieve what you need.
I'll convert this to a discussion, since it's not actually a bug.
